lld/ELF/SyntheticSections.cpp

```diff
@@ -2167,8 +2167,44 @@ void SymbolTableBaseSection::sortSymTabSymbols() {
 
 void SymbolTableBaseSection::addSymbol(Symbol *b) {
   // Adding a local symbol to a .dynsym is a bug.
-  assert(this->type != SHT_DYNSYM || !b->isLocal());
-  symbols.push_back({b, strTabSec.addString(b->getName(), false)});
+  // In SBF, we require all function symbols to be in the dynamic symbol table
+  // regardless of their visibility, so adding local symbols is not a bug for
+  // SBFv3.
+  assert(this->type != SHT_DYNSYM || !b->isLocal() ||
+         (config->emachine == EM_SBF && config->eflags == 0x3));
+
+  unsigned Offset;
+  if (this->type == SHT_DYNSYM && b->isLocal() &&
+      config->strip != StripPolicy::None) {
+    // Including the symbol name in the dynamic symbol table may increase the
+    // contract size in more than 20kb, so whenever we want to strip
+    // information, we can assign the same string to all local symbols.
```

Comment:
The string table for the dynamic symbols would be stripped for production releases anyway, so what difference does it make?

Reply:
The tooling we have from LLVM doesn't work well in ELFs without a dynamic symbol table, because when you have a dynamic symbol table you also expect a dynamic string table. I thought it would be best to maintain it there, and simply add a dummy name to avoid bloating the binary.

Reply:
No, what I meant by production release is that you strip the (dynamic) string table from your ELF before uploading / shipping / deploying it.